syntax = "proto3";

package cafm.identity.user;

import "user.proto";

message UserAppService_userByNameRequest {
  string name = 1;
}

message UserAppService_userByNameResponse {
  User user = 1;
}

message UserAppService_userByIdRequest {
  string id = 1;
}

message UserAppService_userByIdResponse {
  User user = 1;
}

message UserAppService_usersRequest {
  int32 resultFrom = 1;
  int32 resultSize = 2;
}

message UserAppService_usersResponse {
  repeated User users = 1;
}


service UserAppService {
  rpc userByName (UserAppService_userByNameRequest) returns (UserAppService_userByNameResponse) {}
  rpc userById (UserAppService_userByIdRequest) returns (UserAppService_userByIdResponse) {}
  rpc users (UserAppService_usersRequest) returns (UserAppService_usersResponse) {}
}